### PR TITLE
Dispatcher: test cleanup

### PR DESCRIPTION
**Pull request by samspills**
Just removing the extra sleep that was added here: https://github.com/typelevel/cats-effect/pull/3593. The sleep was added as part of a short-term fix for flakiness and now that the second check on alive (and possible cancellation of the submitted task) is gone, this test shouldn't flake. 

I added a `.replicateA(20000)` on line 366 to convince myself that this was stable 🤞🏻

### CODE REVIEW

File: tests/shared/src/test/scala/cats/effect/std/DispatcherSpec.scala [hunk 1]
```diff
@@ -360,7 +360,7 @@ class DispatcherSpec extends BaseSpec with DetectPlatform {
       implicit ticker =>
         val test = dispatcher.allocated.flatMap {
           case (runner, release) =>
-            IO(runner.unsafeRunAndForget(IO.sleep(50.millis) *> release)) *>
+            IO(runner.unsafeRunAndForget(release)) *>
```
Comment (armanbilge) [inline tests/shared/src/test/scala/cats/effect/std/DispatcherSpec.scala:363]: I was briefly confused why we added this sleep in the first place, for a `ticked` test ... and it's because in https://github.com/typelevel/cats-effect/pull/3593 it was originally a `real` test and the sleep helped avoid the race condition where the `release` started while the `unsafeRunAndForget` was still executing.

tl;dr this test is a lot more robust now.

**Review by armanbilge — APPROVED** [review]
Thanks for following-up!
